src/ocean/io/model/ISuspendableThrottler.d

```diff
@@ -77,13 +79,75 @@ abstract public class ISuspendableThrottler
         if ( !this.suspendables.contains(s) )
         {
             this.suspendables ~= s;
-            if (this.suspended_)
+        }
+
+        if (this.suspended_ && !s.suspended())
+        {
+            s.suspend();
+        }
+        else if (!this.suspended_ && s.suspended())
+        {
+            s.resume();
+        }
```

Comment:
Why make it simple if we could make it complicated?

```D
        if (this.suspended_ != s.suspended())
        {
            if (this.suspended_)
                s.suspend();
            else
                s.resume();
        }
```

Reply:
This PR has been merged, but that's a good point.

Reply:
Yep, that's looks indeed better.